twister: fix output path in error messages

AFOliveira opened the pull request:
When twister renames the output directory (e.g., from twister-out to twister-out.7), the error messages still reference files in the original directory path. This happens because TestInstance objects store the original output directory path when they're created, and this path isn't updated when the directory is renamed.

This patch fixes the issue by modifying the log_info method to correct the path in error messages. It detects when a path contains the old output directory and replaces it with the actual output directory path. This ensures that error messages point to files in the correct location, making it easier for users to find log files when debugging.

gchwier commented on scripts/pylib/twister/twisterlib/runner.py:
```diff
@@ -1895,6 +1905,7 @@ def run(self):
                 except queue.Empty:
                     break
                 else:
+                    inst.metrics.update(self.instances[inst.name].metrics)
```
that line was removed: https://github.com/zephyrproject-rtos/zephyr/pull/89945

AFOliveira replied:
Actually, this line was not required, just refactored the logic behind this PR. Thanks for the review.

AFOliveira commented:
To better explain the PR, here's a super verbose output of twister before and after this change:

Before:
```
Renaming output directory to /home/afonsoo/zephyr/**twister-out.34**
INFO    - Using Ninja..
INFO    - Zephyr version: v4.2.0-rc1-251-g980afc8fdfde
INFO    - Using 'cross-compile' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/afonsoo/zephyr/**twister-out**/testplan.json
INFO    - JOBS: 16
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/1 nsim_arc_v/rmx100         samples/hello_world/sample.basic.helloworld        PASSED (nsim 2.728s <cross-compile>)
INFO    -                                    sample.basic.helloworld                                                     PASSED

INFO    - 1 test scenarios (1 configurations) selected, 0 configurations filtered (0 by static filter, 0 at runtime).
Summary
├── Total test suites: 1
├── Processed test suites: 1
│   ├── Filtered test suites: 0
│   │   ├── Filtered test suites (static): 0
│   │   └── Filtered test suites (at runtime): 0
│   └── Selected test suites: 1
│       ├── Skipped test suites: 0
│       ├── Passed test suites: 1
│       ├── Built only test suites: 0
│       ├── Failed test suites: 0
│       └── Errors in test suites: 0
└── Total test cases: 1
    ├── Filtered test cases: 0
    └── Selected test cases: 1
        ├── Passed test cases: 1
        ├── Skipped test cases: 0
        ├── Built only test cases: 0
        ├── Blocked test cases: 0
        ├── Failed test cases: 0
        └── Errors in test cases: 0
INFO    - 1 of 1 executed test configurations passed (100.00%), 0 built (not run), 0 failed, 0 errored, with no warnings in 20.08 seconds.
INFO    - 1 of 1 executed test cases passed (100.00%) on 1 out of total 1115 platforms (0.09%).
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/afonsoo/zephyr/**twister-out**/twister.json
INFO    - Writing xunit report /home/afonsoo/zephyr/**twister-out**/twister.xml...
INFO    - Writing xunit report /home/afonsoo/zephyr/**twister-out**/twister_report.xml...

```
After:
```
Renaming output directory to /home/afonsoo/zephyr/**twister-out.36**
INFO    - Using Ninja..
INFO    - Zephyr version: v4.2.0-rc1-251-g980afc8fdfde
INFO    - Using 'cross-compile' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/afonsoo/zephyr/**twister-out.36**/testplan.json
INFO    - JOBS: 16
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/1 nsim_arc_v/rmx100         samples/hello_world/sample.basic.helloworld        PASSED (nsim 2.769s <cross-compile>)
INFO    -                                    sample.basic.helloworld                                                     PASSED

INFO    - 1 test scenarios (1 configurations) selected, 0 configurations filtered (0 by static filter, 0 at runtime).
Summary
├── Total test suites: 1
├── Processed test suites: 1
│   ├── Filtered test suites: 0
│   │   ├── Filtered test suites (static): 0
│   │   └── Filtered test suites (at runtime): 0
│   └── Selected test suites: 1
│       ├── Skipped test suites: 0
│       ├── Passed test suites: 1
│       ├── Built only test suites: 0
│       ├── Failed test suites: 0
│       └── Errors in test suites: 0
└── Total test cases: 1
    ├── Filtered test cases: 0
    └── Selected test cases: 1
        ├── Passed test cases: 1
        ├── Skipped test cases: 0
        ├── Built only test cases: 0
        ├── Blocked test cases: 0
        ├── Failed test cases: 0
        └── Errors in test cases: 0
INFO    - 1 of 1 executed test configurations passed (100.00%), 0 built (not run), 0 failed, 0 errored, with no warnings in 28.45 seconds.
INFO    - 1 of 1 executed test cases passed (100.00%) on 1 out of total 1115 platforms (0.09%).
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/afonsoo/zephyr/**twister-out.36/**twister.json
INFO    - Writing xunit report /home/afonsoo/zephyr/**twister-out.36**/twister.xml...
INFO    - Writing xunit report /home/afonsoo/zephyr/**twister-out.36**/twister_report.xml...
INFO    - Run completed
```

gchwier commented on scripts/pylib/twister/twisterlib/twister_main.py:
```diff
@@ -62,7 +62,9 @@ def twister(options: argparse.Namespace, default_options: argparse.Namespace):
-                    shutil.move(options.outdir, new_out)
+                    os.makedirs(new_out, exist_ok=True)
+                    # Set options.outdir to the new path before any other operations
+                    options.outdir = new_out
```
You cannot just rename `outdir`, because then every build artifacts will be in that new dir (`twister-out.X`), that is rather not expected (twister-out is the default dir for new build, unless you give `--outdir` .. and it would be more confusing for people, when they define `--outdir` and the output would be in different folder).
Paths in log messages is not a really problem, because you have that path also in twister.json file, which can be used when you re-run the job.
Please check the options:
```
  -c, --clobber-output  Cleaning the output directory will simply delete it
                        instead of the default policy of renaming.
  ...
  -n, --no-clean        Re-use the outdir before building. Will result in
                        faster compilation since builds will be incremental.
  --aggressive-no-clean 
  --test-only, --only-failed ..
  ...
  -O OUTDIR, --outdir OUTDIR
                        Output directory for logs and binaries. Default is
                        'twister-out' in the current directory. This directory
                        will be cleaned unless '--no-clean' is set. The '--
                        clobber-output' option controls what cleaning does.
```

gchwier commented:
> I understand how small the problem is, but still some clarification would probably make the life of folks new to twister (like I am) easier. Could be as small as changing `"Renaming output directory"` to `"Renaming previous output directory",` may I go ahead and do that or do you guys prefer keeping it that way?

sure, you can change it

AFOliveira committed:
scripts: twister: clarify output directory renaming message

Change "Renaming output directory to" message to "Renaming previous
output directory to" to make it clearer that twister is renaming the
existing directory, not the current output directory.

This helps avoid confusion for new twister users who might think the
current output is going to the renamed directory.

Signed-off-by: Afonso Oliveira <afonsoo@synopsys.com>

AFOliveira commented:
> sure, you can change it

Done, thanks for the help figuring this oneout!
